assets/js/lib/filters/mealsMapFilters.js

```diff
@@ -0,0 +1,22 @@
+class MealsMapFilters {
+    constructor(onChangeCallback) {
+      this.parentMapFilters = new MapFilters({
+        filterNames: [
+          "shouldHaveFree",
+        ],
+        filtersElemSelector: ".meals-filters input",
+        onChangeCallback: onChangeCallback
+      });
+  
+      this.anyApplied = this.parentMapFilters.anyApplied;
+    }
+  
+    shouldShowLocation(location) {
+      if (this.parentMapFilters.state.shouldHaveFree && location.hasFree) {
```

Comment:
@Rakshith-2905 Rename `shouldHaveFree` to `shouldHaveFreeMeals` so that it's more explanatory. Please rename in the HTML as well. 

Also, `hasFree` to `hasFreeMeals`.